### PR TITLE
Address some issues with PHPStan level 2

### PR DESCRIPTION
**Pull request by franmomu**
This PR addresses most of the issues found raising PHPStan to level 2 in `tests` directory.

### CODE REVIEW

File: tests/Gedmo/Timestampable/Fixture/Document/Book.php [hunk 1]
```diff
@@ -28,7 +28,7 @@ class Book
     /**
      * @ODM\EmbedMany(targetDocument="Gedmo\Tests\Timestampable\Fixture\Document\Tag")
      *
-     * @var Tag[]|Collection
+     * @var Tag[]|Collection<int, Tag>
```
Comment (phansys) [inline tests/Gedmo/Timestampable/Fixture/Document/Book.php:31]:
```suggestion
     * @var Collection<int, Tag>
```

File: tests/Gedmo/Timestampable/Fixture/Document/Book.php [hunk 2]
```diff
@@ -62,15 +62,15 @@ public function setTitle($title)
...
+     * @return Tag[]|Collection<int, Tag>
      */
     public function getTags()
     {
         return $this->tags;
     }
 
     /**
-     * @param Tag[] $tags
+     * @param Tag[]|Collection<int, Tag> $tags
```
Comment (phansys) [inline tests/Gedmo/Timestampable/Fixture/Document/Book.php:73]:
```suggestion
     * @param Collection<int, Tag> $tags
```

File: tests/Gedmo/Tool/BaseTestCaseORM.php [hunk 3]
```diff
@@ -145,7 +145,7 @@ protected function getMockMappedEntityManager(EventManager $evm = null)
             ->willReturn($this->getMockBuilder(MySQLPlatform::class)->getMock());
 
         $conn = $this->getMockBuilder(Connection::class)
-            ->setConstructorArgs([], $driver)
+            ->setConstructorArgs([[], $driver])
```
Comment (phansys) [inline tests/Gedmo/Tool/BaseTestCaseORM.php:148]: Good catch! :eyes:

**Review by phansys — APPROVED** [review]

File: tests/Gedmo/Mapping/Fixture/Yaml/Category.php [hunk 4]
```diff
@@ -32,14 +34,14 @@ class Category extends BaseCategory
     private $slug;
 
     /**
-     * @var Entity\Category
+     * @var Collection<int, Category>
```
[the remark this answers is not in the document]
Comment (franmomu) [reply tests/Gedmo/Mapping/Fixture/Yaml/Category.php:26]: 👍 I've also removed the annotations which weren't imported since this is used with YAML mapping.

**Pull request merged**